source/directives/activecode.rst

```diff
@@ -0,0 +1,642 @@
+Activecode
+==========
+
+The ``activecode`` directive embeds executable, editable code in your pages. 
+Your students can experiment with your examples by changing them and running them over and over again.
+
+Synopsis
+--------
+The general format of the ``activecode`` directive is:
+
+.. code-block:: rst
+
+   .. activecode:: unique_id
+      :options:
+
+      + --- Content area ---+
+      | question text or additional instructions 
+      | ~~~~
+      | one or more lines of code in a supported language
+      | ====
+      | hidden code to be executed
+      + --------------------+
+
+Depending on the code within the content area, 
+either a pre-formatted text area, or an HTML graphics canvas
+will be created to contain the program output.
+      
+Required Arguments
+------------------
+A unique identifier after a space and the ``::`` in the ``activecode`` directive.
+Valid identifiers must not contain spaces.
+You should also avoid the characters `` ` ``, ``,``, ``:``, and ``*``.
+
+The unique identifier will be the ``div`` id that contains this specific code block. 
+The unique identifier allows you to tie ``activecode`` blocks to a grading interface, 
+or any other groupings for assessment within the Runestone interface. 
+For this reason, we recommend that you follow some type of naming convention 
+for unique identifiers in directives.
+
+Optional Arguments
+------------------
```

Comment:
activecodes have a :timelimit: option that is specified in milliseconds the default is 25 seconds which is longer than should be necessary for most CS1 programs, but can be set shorter or longer using this option

Reply:
done.